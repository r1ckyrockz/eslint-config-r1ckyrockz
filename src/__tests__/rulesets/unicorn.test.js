const { createUnicornRules } = require('../../rulesets/unicorn');

describe('createUnicornRules', () => {
  test('matches snapshot without typescript & react', () => {
    const project = {
      react: {
        hasReact: false,
      },
      typescript: {
        hasTypeScript: false,
      },
    };

    expect(createUnicornRules(project)).toMatchSnapshot();
  });

  test('matches snapshot with typescript, without react', () => {
    const project = {
      react: {
        hasReact: false,
      },
      typescript: {
        hasTypeScript: true,
      },
    };

    expect(createUnicornRules(project)).toMatchSnapshot();
  });

  test('matches snapshot with react, without typescript', () => {
    const project = {
      react: {
        hasReact: true,
      },
      typescript: {
        hasTypeScript: false,
      },
    };

    expect(createUnicornRules(project)).toMatchSnapshot();
  });

  test('matches snapshot with typescript & react', () => {
    const project = {
      react: {
        hasReact: true,
      },
      typescript: {
        hasTypeScript: true,
      },
    };

    expect(createUnicornRules(project)).toMatchSnapshot();
  });

  test('allows passing custom rules', () => {
    const rule = 'unicorn/better-regex';
    const level = 'off';

    const project = {
      react: {
        hasReact: true,
      },
      rules: {
        [rule]: level,
      },
      typescript: {
        hasTypeScript: true,
      },
    };

    const defaultRuleset = createUnicornRules({
      ...project,
      rules: {},
    });

    const result = createUnicornRules(project);

    expect(result[rule]).toBe(level);
    expect(result[rule]).not.toBe(defaultRuleset[rule]);

    expect(result).toMatchSnapshot();
  });
});
