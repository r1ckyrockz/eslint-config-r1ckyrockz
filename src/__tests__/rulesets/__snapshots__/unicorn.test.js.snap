// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createUnicornRules allows passing custom rules 1`] = `
Object {
  "unicorn/better-regex": "off",
  "unicorn/catch-error-name": "error",
  "unicorn/consistent-function-scoping": "error",
  "unicorn/custom-error-definition": "warn",
  "unicorn/error-message": "error",
  "unicorn/escape-case": "error",
  "unicorn/expiring-todo-comments": "off",
  "unicorn/explicit-length-check": "error",
  "unicorn/filename-case": "off",
  "unicorn/import-index": "error",
  "unicorn/new-for-builtins": "error",
  "unicorn/no-abusive-eslint-disable": "error",
  "unicorn/no-array-instanceof": "error",
  "unicorn/no-console-spaces": "error",
  "unicorn/no-fn-reference-in-iterator": "off",
  "unicorn/no-for-loop": "off",
  "unicorn/no-hex-escape": "error",
  "unicorn/no-keyword-prefix": "off",
  "unicorn/no-nested-ternary": "off",
  "unicorn/no-new-buffer": "error",
  "unicorn/no-null": "off",
  "unicorn/no-object-as-default-parameter": "warn",
  "unicorn/no-process-exit": "error",
  "unicorn/no-reduce": "off",
  "unicorn/no-unreadable-array-destructuring": "warn",
  "unicorn/no-unsafe-regex": "error",
  "unicorn/no-unused-properties": "warn",
  "unicorn/no-useless-undefined": "error",
  "unicorn/no-zero-fractions": "error",
  "unicorn/number-literal-case": "off",
  "unicorn/prefer-add-event-listener": "error",
  "unicorn/prefer-array-find": "error",
  "unicorn/prefer-dataset": "error",
  "unicorn/prefer-event-key": "error",
  "unicorn/prefer-flat-map": "error",
  "unicorn/prefer-includes": "off",
  "unicorn/prefer-modern-dom-apis": "error",
  "unicorn/prefer-negative-index": "error",
  "unicorn/prefer-node-append": "error",
  "unicorn/prefer-node-remove": "error",
  "unicorn/prefer-number-properties": "warn",
  "unicorn/prefer-optional-catch-binding": "error",
  "unicorn/prefer-query-selector": "error",
  "unicorn/prefer-reflect-apply": "error",
  "unicorn/prefer-replace-all": "error",
  "unicorn/prefer-set-has": "error",
  "unicorn/prefer-spread": "error",
  "unicorn/prefer-starts-ends-with": "off",
  "unicorn/prefer-string-slice": "error",
  "unicorn/prefer-text-content": "error",
  "unicorn/prefer-trim-start-end": "error",
  "unicorn/prefer-type-error": "error",
  "unicorn/prevent-abbreviations": "off",
  "unicorn/string-content": "off",
  "unicorn/throw-new-error": "error",
}
`;

exports[`createUnicornRules matches snapshot with react, without typescript 1`] = `
Object {
  "unicorn/better-regex": "error",
  "unicorn/catch-error-name": "error",
  "unicorn/consistent-function-scoping": "error",
  "unicorn/custom-error-definition": "warn",
  "unicorn/error-message": "error",
  "unicorn/escape-case": "error",
  "unicorn/expiring-todo-comments": "off",
  "unicorn/explicit-length-check": "error",
  "unicorn/filename-case": "off",
  "unicorn/import-index": "error",
  "unicorn/new-for-builtins": "error",
  "unicorn/no-abusive-eslint-disable": "error",
  "unicorn/no-array-instanceof": "error",
  "unicorn/no-console-spaces": "error",
  "unicorn/no-fn-reference-in-iterator": "off",
  "unicorn/no-for-loop": "warn",
  "unicorn/no-hex-escape": "error",
  "unicorn/no-keyword-prefix": "off",
  "unicorn/no-nested-ternary": "off",
  "unicorn/no-new-buffer": "error",
  "unicorn/no-null": "off",
  "unicorn/no-object-as-default-parameter": "warn",
  "unicorn/no-process-exit": "error",
  "unicorn/no-reduce": "off",
  "unicorn/no-unreadable-array-destructuring": "warn",
  "unicorn/no-unsafe-regex": "error",
  "unicorn/no-unused-properties": "warn",
  "unicorn/no-useless-undefined": "error",
  "unicorn/no-zero-fractions": "error",
  "unicorn/number-literal-case": "off",
  "unicorn/prefer-add-event-listener": "error",
  "unicorn/prefer-array-find": "error",
  "unicorn/prefer-dataset": "error",
  "unicorn/prefer-event-key": "error",
  "unicorn/prefer-flat-map": "error",
  "unicorn/prefer-includes": "warn",
  "unicorn/prefer-modern-dom-apis": "error",
  "unicorn/prefer-negative-index": "error",
  "unicorn/prefer-node-append": "error",
  "unicorn/prefer-node-remove": "error",
  "unicorn/prefer-number-properties": "warn",
  "unicorn/prefer-optional-catch-binding": "error",
  "unicorn/prefer-query-selector": "error",
  "unicorn/prefer-reflect-apply": "error",
  "unicorn/prefer-replace-all": "error",
  "unicorn/prefer-set-has": "error",
  "unicorn/prefer-spread": "error",
  "unicorn/prefer-starts-ends-with": "error",
  "unicorn/prefer-string-slice": "error",
  "unicorn/prefer-text-content": "error",
  "unicorn/prefer-trim-start-end": "error",
  "unicorn/prefer-type-error": "error",
  "unicorn/prevent-abbreviations": "off",
  "unicorn/string-content": "off",
  "unicorn/throw-new-error": "error",
}
`;

exports[`createUnicornRules matches snapshot with typescript & react 1`] = `
Object {
  "unicorn/better-regex": "error",
  "unicorn/catch-error-name": "error",
  "unicorn/consistent-function-scoping": "error",
  "unicorn/custom-error-definition": "warn",
  "unicorn/error-message": "error",
  "unicorn/escape-case": "error",
  "unicorn/expiring-todo-comments": "off",
  "unicorn/explicit-length-check": "error",
  "unicorn/filename-case": "off",
  "unicorn/import-index": "error",
  "unicorn/new-for-builtins": "error",
  "unicorn/no-abusive-eslint-disable": "error",
  "unicorn/no-array-instanceof": "error",
  "unicorn/no-console-spaces": "error",
  "unicorn/no-fn-reference-in-iterator": "off",
  "unicorn/no-for-loop": "off",
  "unicorn/no-hex-escape": "error",
  "unicorn/no-keyword-prefix": "off",
  "unicorn/no-nested-ternary": "off",
  "unicorn/no-new-buffer": "error",
  "unicorn/no-null": "off",
  "unicorn/no-object-as-default-parameter": "warn",
  "unicorn/no-process-exit": "error",
  "unicorn/no-reduce": "off",
  "unicorn/no-unreadable-array-destructuring": "warn",
  "unicorn/no-unsafe-regex": "error",
  "unicorn/no-unused-properties": "warn",
  "unicorn/no-useless-undefined": "error",
  "unicorn/no-zero-fractions": "error",
  "unicorn/number-literal-case": "off",
  "unicorn/prefer-add-event-listener": "error",
  "unicorn/prefer-array-find": "error",
  "unicorn/prefer-dataset": "error",
  "unicorn/prefer-event-key": "error",
  "unicorn/prefer-flat-map": "error",
  "unicorn/prefer-includes": "off",
  "unicorn/prefer-modern-dom-apis": "error",
  "unicorn/prefer-negative-index": "error",
  "unicorn/prefer-node-append": "error",
  "unicorn/prefer-node-remove": "error",
  "unicorn/prefer-number-properties": "warn",
  "unicorn/prefer-optional-catch-binding": "error",
  "unicorn/prefer-query-selector": "error",
  "unicorn/prefer-reflect-apply": "error",
  "unicorn/prefer-replace-all": "error",
  "unicorn/prefer-set-has": "error",
  "unicorn/prefer-spread": "error",
  "unicorn/prefer-starts-ends-with": "off",
  "unicorn/prefer-string-slice": "error",
  "unicorn/prefer-text-content": "error",
  "unicorn/prefer-trim-start-end": "error",
  "unicorn/prefer-type-error": "error",
  "unicorn/prevent-abbreviations": "off",
  "unicorn/string-content": "off",
  "unicorn/throw-new-error": "error",
}
`;

exports[`createUnicornRules matches snapshot with typescript, without react 1`] = `
Object {
  "unicorn/better-regex": "error",
  "unicorn/catch-error-name": "error",
  "unicorn/consistent-function-scoping": "error",
  "unicorn/custom-error-definition": "warn",
  "unicorn/error-message": "error",
  "unicorn/escape-case": "error",
  "unicorn/expiring-todo-comments": "off",
  "unicorn/explicit-length-check": "error",
  "unicorn/filename-case": "off",
  "unicorn/import-index": "error",
  "unicorn/new-for-builtins": "error",
  "unicorn/no-abusive-eslint-disable": "error",
  "unicorn/no-array-instanceof": "error",
  "unicorn/no-console-spaces": "error",
  "unicorn/no-fn-reference-in-iterator": "off",
  "unicorn/no-for-loop": "off",
  "unicorn/no-hex-escape": "error",
  "unicorn/no-keyword-prefix": "warn",
  "unicorn/no-nested-ternary": "off",
  "unicorn/no-new-buffer": "error",
  "unicorn/no-null": "off",
  "unicorn/no-object-as-default-parameter": "warn",
  "unicorn/no-process-exit": "error",
  "unicorn/no-reduce": "off",
  "unicorn/no-unreadable-array-destructuring": "warn",
  "unicorn/no-unsafe-regex": "error",
  "unicorn/no-unused-properties": "warn",
  "unicorn/no-useless-undefined": "error",
  "unicorn/no-zero-fractions": "error",
  "unicorn/number-literal-case": "off",
  "unicorn/prefer-add-event-listener": "error",
  "unicorn/prefer-array-find": "error",
  "unicorn/prefer-dataset": "error",
  "unicorn/prefer-event-key": "error",
  "unicorn/prefer-flat-map": "error",
  "unicorn/prefer-includes": "off",
  "unicorn/prefer-modern-dom-apis": "error",
  "unicorn/prefer-negative-index": "error",
  "unicorn/prefer-node-append": "error",
  "unicorn/prefer-node-remove": "error",
  "unicorn/prefer-number-properties": "warn",
  "unicorn/prefer-optional-catch-binding": "error",
  "unicorn/prefer-query-selector": "error",
  "unicorn/prefer-reflect-apply": "error",
  "unicorn/prefer-replace-all": "error",
  "unicorn/prefer-set-has": "error",
  "unicorn/prefer-spread": "error",
  "unicorn/prefer-starts-ends-with": "off",
  "unicorn/prefer-string-slice": "error",
  "unicorn/prefer-text-content": "error",
  "unicorn/prefer-trim-start-end": "error",
  "unicorn/prefer-type-error": "error",
  "unicorn/prevent-abbreviations": "off",
  "unicorn/string-content": "off",
  "unicorn/throw-new-error": "error",
}
`;

exports[`createUnicornRules matches snapshot without typescript & react 1`] = `
Object {
  "unicorn/better-regex": "error",
  "unicorn/catch-error-name": "error",
  "unicorn/consistent-function-scoping": "error",
  "unicorn/custom-error-definition": "warn",
  "unicorn/error-message": "error",
  "unicorn/escape-case": "error",
  "unicorn/expiring-todo-comments": "off",
  "unicorn/explicit-length-check": "error",
  "unicorn/filename-case": "off",
  "unicorn/import-index": "error",
  "unicorn/new-for-builtins": "error",
  "unicorn/no-abusive-eslint-disable": "error",
  "unicorn/no-array-instanceof": "error",
  "unicorn/no-console-spaces": "error",
  "unicorn/no-fn-reference-in-iterator": "off",
  "unicorn/no-for-loop": "warn",
  "unicorn/no-hex-escape": "error",
  "unicorn/no-keyword-prefix": "warn",
  "unicorn/no-nested-ternary": "off",
  "unicorn/no-new-buffer": "error",
  "unicorn/no-null": "off",
  "unicorn/no-object-as-default-parameter": "warn",
  "unicorn/no-process-exit": "error",
  "unicorn/no-reduce": "off",
  "unicorn/no-unreadable-array-destructuring": "warn",
  "unicorn/no-unsafe-regex": "error",
  "unicorn/no-unused-properties": "warn",
  "unicorn/no-useless-undefined": "error",
  "unicorn/no-zero-fractions": "error",
  "unicorn/number-literal-case": "off",
  "unicorn/prefer-add-event-listener": "error",
  "unicorn/prefer-array-find": "error",
  "unicorn/prefer-dataset": "error",
  "unicorn/prefer-event-key": "error",
  "unicorn/prefer-flat-map": "error",
  "unicorn/prefer-includes": "warn",
  "unicorn/prefer-modern-dom-apis": "error",
  "unicorn/prefer-negative-index": "error",
  "unicorn/prefer-node-append": "error",
  "unicorn/prefer-node-remove": "error",
  "unicorn/prefer-number-properties": "warn",
  "unicorn/prefer-optional-catch-binding": "error",
  "unicorn/prefer-query-selector": "error",
  "unicorn/prefer-reflect-apply": "error",
  "unicorn/prefer-replace-all": "error",
  "unicorn/prefer-set-has": "error",
  "unicorn/prefer-spread": "error",
  "unicorn/prefer-starts-ends-with": "error",
  "unicorn/prefer-string-slice": "error",
  "unicorn/prefer-text-content": "error",
  "unicorn/prefer-trim-start-end": "error",
  "unicorn/prefer-type-error": "error",
  "unicorn/prevent-abbreviations": "off",
  "unicorn/string-content": "off",
  "unicorn/throw-new-error": "error",
}
`;
